app/src/main/java/de/xikolo/controllers/main/SplashActivity.kt

```diff
@@ -48,6 +54,41 @@ class SplashActivity : AppCompatActivity() {
             }
         }
 
+    private fun migrateStorage() {
+        val old = File(FileUtil.getPublicAppStorageFolderPath())
+        val new = File(FileUtil.createStorageFolderPath(StorageUtil.getInternalStorage(this)))
+        val fileCount = FileUtil.folderFileNumber(old)
+
+        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
+        if (!prefs.contains(getString(R.string.preference_storage))) {
```

Comment:
The`ApplicationPreferences` wrapper does not support the `contains` method which checks whether a preference exists. If it does not exists, we can be sure we do not have files stored in the new location, so a migration is triggered. I thought of adding a `fileCount > 0` condition but that would not have cleaned up an empty folder structure. So yes, when the app is freshly installed, a migration takes place but it will not have any effect because there is nothing to migrate.

Reply:
But you could write a method that returns the boolean value you need here for example.